contracts/core/UserOperationLib.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: GPL-3.0
```

Comment:
we do have "util" folder. doesn't it fit there?

Reply:
I think that this library should be part of the core, no?

Reply:
it is part of core (e.g. entrypoint imports "exec")
but you're right in that it is for "non-ep-specific" data, and UserOp (and its lib) are ep-specific.